src/core/storage/state-migrations.ts

```diff
@@ -68,20 +69,41 @@ export async function migrateWorkspaceToGlobalStorage(context: vscode.ExtensionC
 
 export async function migrateTaskHistoryToFile(context: vscode.ExtensionContext) {
 	try {
-		// If the old taskHistory vs code global state is undefined, do nothing
-		const vscodeGlobalStateTaskHistory = await context.globalState.get("taskHistory")
-		if (vscodeGlobalStateTaskHistory === undefined) {
-			return
+		// Get data from both locations
+		const vscodeGlobalStateTaskHistory = (await context.globalState.get("taskHistory")) as HistoryItem[] | undefined
```

Comment:
```suggestion
		const vscodeGlobalStateTaskHistory = context.globalState.get<HistoryItem[] | undefined>("taskHistory")
```
globalState.get doesn't return a promise